### PR TITLE
Fix digest auth http: ContentLength=xxx with Body length 0

### PR DESCRIPTION
**Pull request by icepie**

### CODE REVIEW

**Comment by icepie** [comment]
#729 729

File: digest.go [hunk 1]
```diff
@@ -55,6 +55,17 @@ func (dt *digestTransport) RoundTrip(req *http.Request) (*http.Response, error)
...
+	// Fix http: ContentLength=xxx with Body length 0
+	if req2.Body == nil {
+		req2.ContentLength = 0
+	} else if req2.GetBody != nil {
+		var err error
+		req2.Body, err = req2.GetBody()
+		if err != nil {
+			return nil, err
+		}
+	}
```
Comment (jeevatkm) [inline digest.go:67]: @icepie Can you please add unit test cases?

**Review by jeevatkm — CHANGES_REQUESTED** [review]
@icepie Can you please add unit test cases?

**Comment by icepie** [comment]
> @icepie Can you please add unit test cases?

Ok! I have added d3026f9b3e574db015e63819fd0f72c8bf4b528c

**Review by jeevatkm — APPROVED** [review]
@icepie thanks for adding Test case

**Pull request merged**